src/setup.ts

```diff
@@ -21,21 +21,28 @@ async function setup(): Promise<void> {
   try {
     // inputs from user
     const version: string = core.getInput('version');
+    const url: string = core.getInput('url');
+    const checksum = core.getInput('checksum').toLowerCase();
+
+    // sanity check
+    if (url && !checksum) {
+      throw new Error("user provided url of customized ORAS CLI release but without SHA256 checksum");
+    }
```

Comment:
Should this check be done in `getReleaseInfo`?

Reply:
move it into `getReleaseInfo`